src/tls/openssl/context.h

```diff
@@ -37,20 +37,34 @@ namespace tls
       // Disable renegotiation to avoid DoS
       SSL_CTX_set_options(
         cfg,
-        SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION |
+        SSL_OP_CIPHER_SERVER_PREFERENCE |
+          SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION |
           SSL_OP_NO_RENEGOTIATION);
       SSL_set_options(
         ssl,
-        SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION |
+        SSL_OP_CIPHER_SERVER_PREFERENCE |
+          SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION |
           SSL_OP_NO_RENEGOTIATION);
 
       // Set cipher for TLS 1.2 (TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256)
       SSL_CTX_set_cipher_list(cfg, "ECDHE-ECDSA-AES128-GCM-SHA256");
       SSL_set_cipher_list(ssl, "ECDHE-ECDSA-AES128-GCM-SHA256");
 
       // Set cipher for TLS 1.3 (same as above)
-      SSL_CTX_set_ciphersuites(cfg, "TLS_AES_128_GCM_SHA256");
-      SSL_set_ciphersuites(ssl, "TLS_AES_128_GCM_SHA256");
+      SSL_CTX_set_ciphersuites(
+        cfg,
+        "TLS_AES_256_GCM_SHA384:"
+        "TLS_CHACHA20_POLY1305_SHA256:"
+        "TLS_AES_128_GCM_SHA256");
+      SSL_set_ciphersuites(
+        ssl,
+        "TLS_AES_256_GCM_SHA384:"
+        "TLS_CHACHA20_POLY1305_SHA256:"
+        "TLS_AES_128_GCM_SHA256");
+
+      // Restrict the curves to approved ones
+      SSL_CTX_set1_curves_list(cfg, "P-521:P-384");
+      SSL_set1_curves_list(ssl, "P-521:P-384");
```

Comment:
Is there an authoritative list to determine this? Perhaps https://docs.microsoft.com/en-us/windows/win32/secauthn/tls-cipher-suites-in-windows-server-2022 or https://csrc.nist.gov/publications/detail/sp/800-52/rev-2/final?